Client/Frontend/BraveVPN/BraveVPN.swift

```diff
@@ -290,6 +290,34 @@ class BraveVPN {
     /// Connects to Guardian's server to validate locally stored receipt.
     /// Returns true if the receipt expired, false if not or nil if expiration status can't be determined.
     static func validateReceipt(receiptHasExpired: ((Bool?) -> Void)? = nil) {
+        /// A helper struct to deal with unwrapping receipt json and extracting properites we are interested in.
+        struct _Receipt: Decodable {
+            let expireDate: Date
+            let isTrialPeriod: Bool?
+
+            private enum CodingKeys: String, CodingKey {
+                case expiresDateMs = "expires_date_ms"
+                case isTrialPeriod = "is_trial_period"
+            }
+            
+            init(from decoder: Decoder) throws {
+                let container = try decoder.container(keyedBy: CodingKeys.self)
+                let expireDateString = try container.decode(String.self, forKey: .expiresDateMs)
```

Comment:
These are all strings in JSON? :(

Reply:
Yes, all values from the receipt we get are strings unfortunately.
We rely on the same server side logic as Guardian team, requesting change for it is not straightforward